api/routes/apds/post.js

```diff
@@ -1,40 +1,82 @@
+const Ajv = require('ajv');
+const moment = require('moment');
+
 const logger = require('../../logger')('apds route post');
 const { raw } = require('../../db');
-const { apd: defaultApdModel } = require('../../db').models;
 const { can } = require('../../middleware');
 
-const defaultGetNewApd = require('./post.data');
+const getNewApd = require('./post.data');
+
+const apdSchema = require('../../schemas/apd.json');
+
+const ajv = new Ajv({
+  allErrors: true,
+  jsonPointers: true,
+  removeAdditional: true
+});
 
-module.exports = (
-  app,
-  { db = raw, getNewApd = defaultGetNewApd, ApdModel = defaultApdModel } = {}
-) => {
+const validatorFunction = ajv.compile({
+  ...apdSchema,
+  additionalProperties: false
+});
+
+module.exports = (app, { db = raw } = {}) => {
   logger.silly('setting up POST /apds/ route');
   app.post('/apds', can('edit-document'), async (req, res) => {
     logger.silly(req, 'handling POST /apds route');
 
     try {
-      const newApd = ApdModel.forge({
-        state_id: req.user.state,
-        status: 'draft'
-      });
-      await newApd.save();
+      const apd = getNewApd();
 
-      const apdContent = await getNewApd(req.user.state);
-      await newApd.synchronize(apdContent);
+      apd.name = `${req.user.state.toUpperCase()}-${moment(Date.now()).format(
+        'YYYY-MM-DD'
+      )}-HITECH-APD`;
 
-      const apd = await ApdModel.where({ id: newApd.get('id') }).fetch({
-        withRelated: ApdModel.withRelated
-      });
+      const stateProfile = await db('states')
+        .select('medicaid_office')
+        .where({ id: req.user.state })
+        .first();
 
-      await db('apds')
-        .where({ id: newApd.get('id') })
-        .update({ document: apd.toJSON() });
+      if (stateProfile) {
+        // Merge the state profile from the states table into the default
+        // values so that if the states table info is missing any fields,
+        // we preserve the defaults
 
-      res.send(apd);
+        apd.stateProfile.medicaidDirector = {
+          ...apd.stateProfile.medicaidDirector,
+          ...stateProfile.medicaid_office.medicaidDirector
+        };
+
+        apd.stateProfile.medicaidOffice = {
+          ...apd.stateProfile.medicaidOffice,
+          ...stateProfile.medicaid_office.medicaidOffice
+        };
+        delete apd.stateProfile.medicaidOffice.director;
```

Comment:
```suggestion
        // An old version of the model had the director info contained inside the office field, so
        // just in case we're still hitting a really old source, delete the director from the office
        delete apd.stateProfile.medicaidOffice.director;
```